libs/langchain/langchain/document_loaders/base.py

```diff
@@ -93,3 +94,44 @@ def parse(self, blob: Blob) -> List[Document]:
             List of documents
         """
         return list(self.lazy_parse(blob))
+
+
+class BaseBytesParser(ABC):
```

Comment:
Hi @esiiol thank you for the contribution -- take a look if this can be used for your use case: 

Blob object which supports loading entirely from memory using bytes objects: https://github.com/langchain-ai/langchain/blob/68599d98c20b1e3fbdabaef1b1fbe54cd06b98a4/libs/langchain/langchain/document_loaders/blob_loaders/schema.py#L21-L21

https://github.com/langchain-ai/langchain/blob/68599d98c20b1e3fbdabaef1b1fbe54cd06b98a4/libs/langchain/langchain/document_loaders/base.py#L57-L57


And implementations of pdf parsers

https://github.com/langchain-ai/langchain/blob/68599d98c20b1e3fbdabaef1b1fbe54cd06b98a4/libs/langchain/langchain/document_loaders/parsers/pdf.py#L253-L253

Reply:
Hi @eyurtsev thanks for your response. That's absolutely right. I'd missed them. I'll update the code accordingly and commit to the PR.